src/Strategy/Erlang/Rebar3Tree.hs

```diff
@@ -0,0 +1,156 @@
+module Strategy.Erlang.Rebar3Tree
+  ( discover
+  , analyze
+
+  , buildGraph
+  , rebar3TreeParser
+  , Rebar3Dep(..)
+  )
+  where
+
+import Prologue
+
+import Control.Carrier.Error.Either
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+import Text.Megaparsec
+import Text.Megaparsec.Char
+
+import DepTypes
+import Discovery.Walk
+import Effect.Exec
+import Graphing (Graphing)
+import Effect.Grapher
+import Types
+
+discover :: HasDiscover sig m => Path Abs Dir -> m ()
+discover = walk $ \dir _ files -> do
+  case find (\f -> fileName f `elem` ["rebar.config"]) files of
+    Nothing -> pure ()
+    Just _  -> runSimpleStrategy "erlang-rebar3tree" ErlangGroup $ analyze dir
+
+  pure WalkContinue
+
+rebar3TreeCmd :: Command
+rebar3TreeCmd = Command
+  { cmdNames = ["rebar3"]
+  , cmdBaseArgs = ["tree", "-v"]
+  , cmdAllowErr = Never
+  }
+
+analyze :: (Has Exec sig m, Has (Error ExecErr) sig m) => Path Rel Dir -> m ProjectClosureBody
+analyze dir = mkProjectClosure dir <$> execParser rebar3TreeParser dir rebar3TreeCmd []
+
+mkProjectClosure :: Path Rel Dir -> [Rebar3Dep] -> ProjectClosureBody
+mkProjectClosure dir deps = ProjectClosureBody
+  { bodyModuleDir    = dir
+  , bodyDependencies = dependencies
+  , bodyLicenses     = []
+  }
+  where
+  dependencies = ProjectDependencies
+    { dependenciesGraph    = buildGraph deps
+    , dependenciesOptimal  = NotOptimal
+    , dependenciesComplete = NotComplete
+    }
+
+type RebarGrapher = LabeledGrapher Rebar3Dep RebarLabel
+
+data RebarLabel =
+    RebarSource Text -- location
```

Comment:
this label isn't actually used -- so let's not use `LabeledGrapher` here. can we use `unfold` instead for `buildGraph`?

Reply:
So I implemented this and I can't get the dependency graph construction to work for dependencies greater than 1 level. I think this may have been a known error and is one of the reasons I went with the `LabeledGrapher` to begin with. Is this familiar with you? I can't find an example where `unfold` is being used with deps deeper than 1 level.

Reply:
From the code, It appears that the `unfold` function only handles deep deps for direct dependencies. It feel like the best path forward here would be to maintain the labeled grapher as it can handle deep deps and the locations that will be found once we can correctly handle `git` dependencies.